### PR TITLE
added build only filter support to error list

### PR DESCRIPTION
**Pull request by heejaechang**
now user has a easy way to see just errors from build. the content should be exactly same as what output build window shows.

this change set also includes several misc changes such as adding support for error source and build tool.

### CODE REVIEW

**Comment by heejaechang** [comment]
fix #2310

**Comment by heejaechang** [comment]
it is failed since Jenkins doesnt have patched dlls

**Comment by heejaechang** [comment]
@shyamnamboodiripad @mavasani @srivatsn @JohnHamby @jmarolf can you take a look please?

File: src/VisualStudio/Core/Def/Implementation/AnalyzerDependencyCheckingService.cs [hunk 1]
```diff
@@ -51,6 +51,11 @@ internal sealed class AnalyzerDependencyCheckingService
                 return;
             }
 
+            if (results == null)
```
Comment (heejaechang) [inline src/VisualStudio/Core/Def/Implementation/AnalyzerDependencyCheckingService.cs]: @tmeschter this is just to stop crashing. you probably need to fix it properly.

Comment (tmeschter) [inline src/VisualStudio/Core/Def/Implementation/AnalyzerDependencyCheckingService.cs]: @heejaechang Looks good to me. Thanks!

**Comment by jmarolf** [comment]
:+1: obviously figure out why IErrorList can't be found, but looks good.

**Comment by heejaechang** [comment]
the test failure is due to Jenkins not patched. I just patched Jenkins.

**Comment by JohnHamby** [comment]
I sign off, but I probably don't understand this code well enough for my signoff to count. A lot of code moves around in this change set, and I couldn't identify the new logic. I did read the code.

**Comment by heejaechang** [comment]
@JohnHamby ya. this required many small refactoring to put in the new functionality.

ExternalErrorDiagnosticUpdateSource.cs and VisualStudioDiagnosticListTable.cs are the main changes though.

**Pull request merged**